users/service.go

```diff
@@ -120,16 +120,22 @@ func (svc service) ViewClient(ctx context.Context, token, id string) (mgclients.
 		return mgclients.Client{}, err
 	}
 
+	client, err := svc.clients.RetrieveByID(ctx, id)
+	if err != nil {
+		return mgclients.Client{}, errors.Wrap(svcerr.ErrViewEntity, err)
+	}
+
 	if tokenUserID != id {
 		if err := svc.checkSuperAdmin(ctx, tokenUserID); err != nil {
-			return mgclients.Client{}, err
+			client, clientErr := svc.clients.RetrieveByID(ctx, id)
```

Comment:
`RetrieveByID` is being called twice. No need for this you can refactor it to something like

```go
client, err := svc.clients.RetrieveByID(ctx, id)
if err != nil {
	return mgclients.Client{}, errors.Wrap(svcerr.ErrViewEntity, err)
}

if tokenUserID != id {
	if err := svc.checkSuperAdmin(ctx, tokenUserID); err != nil {
		return mgclients.Client{Name: client.Name, ID: client.ID}, nil
	}
}

client.Credentials.Secret = ""

```